Content.Server/Kitchen/EntitySystems/ReagentGrinderSystem.cs

```diff
@@ -81,6 +81,15 @@ public override void Update(float frameTime)
 
                 foreach (var item in inputContainer.ContainedEntities.ToList())
                 {
+                    // Frontier code for regent grinder tier 2 - Start
+                    TryComp<ExtractableComponent>(item, out var extractable);
+                    if (!reagentGrinder.Advanced && extractable!.Advanced)
+                    {
+                        _popupSystem.PopupEntity(Loc.GetString("advanced-reagent-grinder-component-cannot-grind-entity-message"), uid, PopupType.MediumCaution);
+                        continue;
+                    }
+                    // Frontier code for regent grinder tier 2 - End
+
```

Comment:
Good luck doing this without a full rewrite of the grinder code since this isnt build around it at all,
Allowed to be inserted into grinder is REGENT IN ITEM BASED not whitelist / blacklist.